### PR TITLE
Port ScrollDAOTable from mdittmer/mdn-confluence

### PR DESCRIPTION
**Pull request by mdittmer**
ScrollDAOTable decorates a scrolling table view with table headers, and, optionally, "X" buttons for removing columns. The view has a manual UI test.

### CODE REVIEW

File: lib/u2/ScrollDAOTable.es6.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+      methods: [
+        function format(data, opt_columns) {
+          const columns = opt_columns || this.columns;
+
+          let str = '';
+          for (let i = 0; i < columns.length; i++) {
+            const col = columns[i];
+            const value = data ? col.f(data) : undefined;
+            str += col.rawTableCellFormatter(data ? col.f(data) : undefined, data,
```
Comment (arobins) [inline lib/u2/ScrollDAOTable.es6.js]: Did you mean to use value here?

File: lib/u2/ScrollDAOTable.es6.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  name: 'ScrollDAOTable',
+  package: 'org.chromium.apis.web',
+  extends: 'foam.u2.View',
+
+  documentation: `Decorator for foam.u2.view.ScrollDAOView that adds column
+      headers, and the ability to remove columns witha "X" button per column.`,
```
Comment (arobins) [inline lib/u2/ScrollDAOTable.es6.js]: nit: with a

File: lib/u2/ScrollDAOTable.es6.js (hunk 1 above)
Comment (arobins) [inline lib/u2/ScrollDAOTable.es6.js]: What happens if there is no formatter?

File: lib/u2/ScrollDAOTable.es6.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      name: 'of',
+      // TODO(markdittmer): This should be an expression, but that's not
+      // working.
+      factory: function() { return this.data.of; },
+    },
+  ],
+
+  methods: [
+    function initE() {
+      const self = this;
```
Comment (arobins) [inline lib/u2/ScrollDAOTable.es6.js]: nit: appears unused

File: lib/u2/ScrollDAOTable.es6.js (hunk 2 above)
Comment (mdittmer) [reply lib/u2/ScrollDAOTable.es6.js]: Done.

File: lib/u2/ScrollDAOTable.es6.js (hunk 1 above)
Comment (mdittmer) [reply lib/u2/ScrollDAOTable.es6.js]: Yes. Done.

Comment (mdittmer) [reply lib/u2/ScrollDAOTable.es6.js]: Good point. I added a safe default to `property.es6.js`.

File: lib/u2/ScrollDAOTable.es6.js (hunk 3 above)
Comment (mdittmer) [reply lib/u2/ScrollDAOTable.es6.js]: Used on lines 192, 193

Comment (arobins) [reply lib/u2/ScrollDAOTable.es6.js]: Ok. It's in a block that also uses this. Why have two names for the same thing?

File: test/manual/ScrollDAOTable.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    <script type="text/javascript" src="../../static/bundle/vendors.bundle.js"></script>
+    <script type="text/javascript" src="../../static/bundle/foam.bundle.js"></script>
+    <script type="text/javascript" src="../../static/bundle/ui_test.bundle.js"></script>
+  </head>
+  <body>
+    <div style="padding: 16px">
+      Expectations:
+      <ul>
+        <li>
+          Scrollabele Checkerboard grid of API names + 4 fake browser
```
Comment (arobins) [inline test/manual/ScrollDAOTable.html]: nit: scrollable

**Pull request merged**